### PR TITLE
Update iOS SDK to 16.2.3

### PR DESCRIPTION
**Pull request by Br1an-Boyle**
- Update Intercom iOS SDK to 16.2.3
- Bumped Android SDK compile version to 34.

### CODE REVIEW

File: intercom_flutter/android/build.gradle [hunk 1]
```diff
@@ -25,7 +25,7 @@ apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 
 android {
-    compileSdkVersion 33
+    compileSdkVersion 34
```
Comment (Zazo032) [inline intercom_flutter/android/build.gradle:28]:
```suggestion
    compileSdk 34
```

File: intercom_flutter/README.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ Flutter wrapper for Intercom [Android](https://github.com/intercom/intercom-andr
 - Uses Intercom Android SDK Version `15.4.0`.
 - The minimum Android SDK `minSdk` required is 21.
 - The compile Android SDK `compileSdk` required is 34.
-- Uses Intercom iOS SDK Version `16.2.1`.
+- Uses Intercom iOS SDK Version `16.2.13`.
```
Comment (Zazo032) [inline intercom_flutter/README.md]:
```suggestion
- Uses Intercom iOS SDK Version `16.2.3`.
```

**Review by Zazo032 — APPROVED** [review]
LGTM with 2 small nits. Thanks!

File: intercom_flutter/android/build.gradle (hunk 1 above)
Comment (deepak786) [reply intercom_flutter/android/build.gradle:28]: We can apply this change later as there is also minSdkVersion which will be minSdk

**Review by deepak786 — APPROVED** [review]

**Pull request merged**